WordPress/gutenberg
# Core Data: Support entity queries in the 'useResourcePermissions' hook

## What?
This is a follow-up to #63322.

PR updates the `useResourcePermissions` hook to support new entity-aware syntax. I also replaced usage in the codebase.

## Why?
The new syntax should be preferred over direct REST API resource queries.

## Testing Instructions

1. Go to the Navigation block
2. Create a new item in the canvas and type in some random text to generate a new page.
3. It should be possible to create a page.
4. Do the same but using a Post Link block in the Nav.

### Testing Instructions for Keyboard
Same.

## Screenshots or screencast 
![CleanShot 2024-07-17 at 16 14 25](https://github.com/user-attachments/assets/8e15943d-b4b4-432d-8aea-3c3b7f38e40e)
